account_ecotax_sale/views/sale_view.xml

```diff
@@ -0,0 +1,157 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!--
+   Copyright (C) 2015-2023 Akretion (http://www.akretion.com/)
+   @author: Mourad EL HADJ MIMOUNE <mourad.elhadj.mimoune@akretion.com>
+-->
+<odoo>
+    <record id="view_order_form" model="ir.ui.view">
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_form" />
+        <field name="arch" type="xml">
+            <field name="tax_totals" position="after">
+                <field name="amount_ecotax" class="oe_subtotal_footer_separator" />
+            </field>
+            <xpath
+                expr="//field[@name='order_line']/form//field[@name='tax_id']"
+                position="after"
+            >
+                <field name="ecotax_line_ids" colspan="2">
+                    <tree editable="bottom">
+                        <field name="classification_id" />
+                        <field name="force_amount_unit" />
+                        <field name="amount_unit" />
+                        <field name="amount_total" />
+                    </tree>
+                </field>
+                <field name="subtotal_ecotax" />
+                <field name="ecotax_amount_unit" />
+            </xpath>
+
+            <xpath
+                expr="//field[@name='order_line']/tree/field[last()]"
+                position="after"
+            >
+                <field name="subtotal_ecotax" optional="show" />
+                <field name="ecotax_amount_unit" optional="hide" />
+                <button name="edit_ecotax_lines" type="object" string="Ecotax" />
+                <field name="ecotax_line_ids" invisible="1">
+                    <!-- This is important to have these fields in view for the computed readonly False ecotax_line_ids field compute well
+                    as it acts as an onchange. -->
+                    <tree>
+                        <field name="classification_id" />
+                        <field name="force_amount_unit" />
+                        <field name="amount_unit" />
+                        <field name="amount_total" />
+                    </tree>
+                </field>
+            </xpath>
+        </field>
+    </record>
+
+    <record id="view_sale_line_ecotax_form" model="ir.ui.view">
+        <field name="name">sale.order.line.ecotax.form</field>
+        <field name="model">sale.order.line</field>
+        <field name="priority" eval="1000" />
+        <field name="arch" type="xml">
+            <form create="false">
+                <sheet>
+                    <field name="display_type" invisible="1" />
+                    <!--
+                        We need the sequence field to be here for new lines to be added at the correct position.
+                        TODO: at some point we want to fix this in the framework so that an invisible field is not required.
```

Comment:
could this migration be the opportunity to resolve this TODO? :crossed_fingers:

Reply:
@vvrossem this TODO is actually copypasted from the original SO line form view from `sale` module: https://github.com/odoo/odoo/blob/a848c3854c94b5c2b752edddbcf49337acf9d6ea/addons/sale/views/sale_order_views.xml#L366-L373